### ISSUE TITLE
Store doesn't persist after rebuilding it when logging out

### ISSUE THREAD

**Issue by high-performance-hery** [morrys/react-relay-offline#111]
When logging out, we rebuild a new environment like this:

        const store = new Store(recordSource, persistOptionsStore, relayStoreOptions)
        const env = new Environment(
            {
                network,
                store,
            },
            {}
        )

but the issue is that the store doesn't persist any data after that, e.g. mutations don't mutate anything until we hard restart the app, then it works normally.

Is there anything specific we need to do when rebuilding the environment for the store to persist data properly?

**Comment by morrys** [comment]
Hi @high-performance-hery,
It appears that the restore of the new environment is not performed

**Comment by high-performance-hery** [comment]
We do call `useRestore` with the newly built environment, is there anything else we need to do?

**Comment by morrys** [comment]
Add the log to understand if the new environment has been restored and if the mutation is performed in the new enrivonment

**Comment by morrys** [comment]
Otherwise, create a small sample project where the problem occurs so that I can help you

**Comment by high-performance-hery** [comment]
So the new environment has been restored and the mutation is performed in the new environment.

However it looks like the first mutation updates the store correctly, but the UI doesn't get updated. So subsequent mutations just send the same mutation as the initial one and no more updates occur.

After restarting the app, mutations behave normally.

So it looks like we set up the Relay store and environment correctly, but there's an issue with the way we hook up the UI?

**Comment by morrys** [comment]
I tried to modify the example project by adding a button that makes a new "login" and everything seems to work correctly.
https://github.com/morrys/offline-examples/blob/master/relay/todo-updater/js/app.js#L106

**Comment by high-performance-hery** [comment]
Looks good, thanks! It looks like a setup issue on our end: queries go to the old store and mutations go to the new one, which is why the UI doesn't update when the store updates.

**Comment by high-performance-hery** [comment]
Do I need to update the RelayEnvironmentProvider on logout? Doing this via a state update is tricky, since there is a place where I need to use our logout function imperatively (i.e. outside a component render function). Previously I had just been clearing async storage/access token and running buildEnvironment. Is there a workaround in this case?

**Comment by morrys** [comment]
Yes you have to update it, otherwise the application does not log out completely, in fact it uses the enviroment associated with the old user.

It is difficult for me to give you advice on how to do this in your case.

**Comment by high-performance-hery** [comment]
Yea sounds quite specific to our setup.

Would there be a problem with updating the env when logging in or signing up rather than logging out?

**Comment by morrys** [comment]
No problem, but you must avoid that the application uses the old environment in the meantime
